int64 mode
---
int64 result
